hack/ginkgo-e2e.sh

```diff
@@ -152,7 +159,17 @@ CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-${KUBE_CONTAINER_RUNTIME:-}}
 # Add path for things like running kubectl binary.
 PATH=$(dirname "${e2e_test}"):"${PATH}"
 export PATH
-"${ginkgo}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
+
+# Choose the program to execute.
+program=("${ginkgo}")
+if [[ "${E2E_TEST_DEBUG_TOOL}" == "delve" ]]; then
+  program=("dlv" "exec")
+fi
+if [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
+  program=("gdb")
+fi
```

Comment:
Typically I would expect this sort of pattern to be a switch/case. Maybe this is fewer lines though. Could also reduce by changing the middle to `elif`

Reply:
changed it to `elif`